### DISCUSSION TITLE
Differences among bin files

### DISCUSSION THREAD

**Discussion by DDMH** [moonglow/FlashForge_Marlin#43] [Q&A] [answered]
What are the differences among the bin files on a particular machine? For example, the Color UI zip folder includes the following files for a Flashforge Dreamer:

fw_2.0.9.1_dreamer_06302021.bin
fw_2.0.9.1_dreamer_la_06302021.bin
fw_2.0.9.1_dreamer_swap_06302021.bin
fw_2.0.9.1_dreamer_swap_la_06302021.bin

I think an explanation should be included in the Wiki or Release Notes. Thanks!

Edit: Is the la_ version linear advance enabled? Still haven't figured out about _swap.

**Comment by moonglow** [comment, marked as answer]
This is only my pre built setup, source code is open and people can build any combination of millions options by yourself, but i chose some of useful templates for my release page:

> Edit: Is the la_ version linear advance enabled? Still haven't figured out about _swap.

Yep !

**SWAP** - by default ( as original firmware do that too ) your left hotend has index 1 and right one has index 0 this is illogical for many peoples, so **SWAP** firmware change it to more logical way, left hotend will have index 0 and right one index 1.

**Reply by DDMH** [reply]
Awesome. I will flash mine using the swap version.

**Comment by ammoniak** [comment]
What does the "dir" stand for? (all the other are clear after reading the build-scripts's comments ;))

**Reply by moonglow** [reply]
@ammoniak 

>What does the "dir" stand for?

Some old Dreamer motherboards has opposite direction control for second extrude... it like swap, but swaps only ONE extruder... and looks like i forgot to update my build script ^_^
If one of your extruder is move filament to the right way and second one to opposite you need to use _dir version, or manually change direction of wrong extruder by yourself if you build your own firmware by hands.

**Comment by ammoniak** [comment]
Thank you. I'm currently using your builds. (tried the "normal" one first, but I'm gonna test "swap" + "linear advance" next. I'll see quickly if "dir" is required as well ;)

Thanks for the time you invested into this project. I'm glad I can use Marlin now :) It makes live so much easier

**Comment by ammoniak** [comment]
I just loaded `dreamer_2.0.9.2_dir_swap_la_10042021` and `dreamer_2.0.9.2_swap_la_10042021` and on both the second (right) extruder extrudes in the wrong direction :(

**Reply by ammoniak** [reply]
(note: the first/left one works properly with both versions)

**Reply by moonglow** [reply]
yeh... sorry i just forget to update sources ( now it was updated ) i will make new builds tomorrow to fix that issue. Previous release was mades from unedited sources ( If you can build it yourself ( build script was updated too ) let me know

**Reply by ammoniak** [reply]
12.3 works fine. I can't build it on my windows machine and don't have platformio installed on linux. I'll change that soon ;)

**Reply by ammoniak** [reply]
alright. just installed platformio on my linux and it compiles / builds /etc. I will flash it today or tomorrow (as soon as I get to it) and check if everything works.

**Reply by moonglow** [reply]
@ammoniak all is OK with updated sources ?

**Reply by ammoniak** [reply]
I got lost in the linear advance calibration, I'm still on 12.3. I'll reflash this evening.

**Reply by ammoniak** [reply]
Alright, I flashed 2.0.9.2 (the version I built my self) and it does not work:
- swap + dir => both extruder turn in the wrong direction
- swap => one extruder turns in the wrong direction

**Reply by moonglow** [reply]
Thank you ! I will invert logic to correct way

**Reply by moonglow** [reply]
@ammoniak i merged new Marlin updates and change pin configuration ( revert direction logic... yep again... ), if all be OK i will update repo. 
Can you check it ? [LINK](https://github.com/moonglow/FlashForge_Marlin/releases/tag/0.13.1)

p.s: because swap + invert drive me crazy  ^_^ and i do not have real hardware to test that case :(

**Reply by ammoniak** [reply]
I can confirm: dreamer_2.0.9.2_dir_swap_11252021.bin works on my printer.

**Reply by moonglow** [reply]
@ammoniak Thank you! Repo was updated with that fixes
